mad_prefect/data_assets/data_artifact.py

```diff
@@ -210,7 +213,9 @@ async def _persist_csv(self):
                 # If it's already a pa.Table or pa.RecordBatch, use it.
                 # Otherwise, convert it to a Table from a list-of-dicts or list-of-rows.
                 if not isinstance(next_entity, (pa.Table, pa.RecordBatch)):
-                    next_entity = pa.Table.from_pylist(next_entity)
+                    next_entity = pa.Table.from_pylist(
+                        TypeAdapter(Any).dump_python(next_entity)
```

Comment:
Although the TypeAdapter(Any) isn't as expensive as creating other TypeAdapters, it is recommended that we create a single TypeAdapter to reuse across the batches, rather than creating a TypeAdapter instance for every batch.